src/main/java/com/yelp/nrtsearch/server/luceneserver/doc/VectorType.java

```diff
@@ -0,0 +1,79 @@
+/*
+ * Copyright 2022 Yelp Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.yelp.nrtsearch.server.luceneserver.doc;
+
+import java.lang.reflect.Array;
+import java.util.AbstractList;
+
+public final class VectorType extends AbstractList<Float> {
+
+  private static final int DEFAULT_CAPACITY = 10;
+
+  private float[] vectorData;
+
+  private int vectorSize;
```

Comment:
Since there is not resizing anymore, you can probably just use the array size now.

Reply:
I have kept this variable as is since set() method is now added.